lib/strings.nix

```diff
@@ -756,7 +756,14 @@ rec {
        sanitizeDerivationName pkgs.hello
        => "-nix-store-2g75chlbpxlrqn15zlby2dfh8hr9qwbk-hello-2.10"
   */
-  sanitizeDerivationName = string: lib.pipe string [
+  sanitizeDerivationName =
+  let okRegex = match "^[[:alnum:]+_?=-][[:alnum:]+-._?=]*$";
```

Comment:
This regex has a mistake in it! Check this out:
```
nix-repl> strings.sanitizeDerivationName "a,"    
"a,"
```

Whereas before this PR:

```
nix-repl> strings.sanitizeDerivationName "a,"    
"a-"
```

<details>
<summary>Expand for spoilers why this happens (and the solution):</summary>

For the regex engine Nix is using, `-` in brackets has a different meaning depending on whether it's first/last or in the middle:
- If `-` occurs in the middle, like in `[[:alnum:]+-._?=]`, then it means the ascii range from the preceding to the succeeding character, which in this case is from `+` to `.`, which notably includes `+`, `,`, `-` and `.`!
- If `-` occurs at the beginning or the end, like in `[[:alnum:]+_?=-]`, then it means only `-` itself

So to fix this, the `-` needs to be moved to e.g. the end of the second `[]`
</details>

Reply:
Nice attention to `<details>`!

Did I mention I hate regex in its usual form? ("usual" because they're impossible to hate in their theoretical form)

<details><summary>half-way spoiler</summary>

I took the characters from the docs or whatever, noticed the problem, then proceeded to fix it once instead of twice :face_with_head_bandage: 

</details>